query-planner-js/src/buildPlan.ts

```diff
@@ -1138,13 +1174,49 @@ class FetchGroup {
     const inputNodes = inputs ? inputs.toSelectionSetNode() : undefined;
 
     const subgraphSchema = this.dependencyGraph.subgraphSchemas.get(this.subgraphName)!;
-    let operation = this.isEntityFetch
-      ? operationForEntitiesFetch(
-          subgraphSchema,
-          this.selection,
-          variableDefinitions,
+    let operation: Operation;
+    const finders = this.isEntityFetch ? this.getFindersForInput(inputs) : [];
+
+    if (finders.length > 0) {
+      // if multiple finders are found, just use the first one for now
+      const finder = finders[0];
+      assert(finder, 'operationForFindersFetch called without a finder');
+      const operation = operationForFindersFetch({
+        subgraphSchema: subgraphSchema,
+        selectionSet: this.selection,
+        allVariableDefinitions: variableDefinitions,
+        operationName,
+        finder,
+      });
+      const operationDocument = operationToDocument(operation);
+      const subgraphFetchNode: SubgraphFetchNode = {
+        kind: 'SubgraphFetch',
+        serviceName: this.subgraphName,
+        id: this.id,
+        operationName: operation.name ?? 'finder call',
+        operation: stripIgnoredCharacters(print(operationDocument)),
+        variableUsages: [finder.fieldName()],
+        inputs: [{
+          selectedType: finder.argument(0).inputType.toString(),
```

Comment:
Fwiw, neither `selectedType` nor `variableName` seems to be used during execution at all, so unclear what their purpose is.